src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -469,4 +457,30 @@ private String evaluateMacro(String template) {
             throw new IllegalArgumentException(e);
         }
     }
+
+    private void pullRequestActionable(List<PotentialAction> paList, List<Fact> factsList) {
+        Job job = run.getParent();
+        ItemGroup parent = job.getParent();
+        if (parent instanceof MultiBranchProject) {
```

Comment:
@stephenc was this how you intended Branch API to be used? 👍 
I considered just trying to fetch the environment variables, provided by https://github.com/jenkinsci/branch-api-plugin/blob/4ba1f7f965071dfc4fe6a416a48283a4a001cc91/src/main/java/jenkins/branch/BranchNameContributor.java#L54-L95

Reply:
So much better would be 
```
SCMHead head = SCMHead.HeadByItem.findHead(job)
if (head instanceof ChangeRequestSCMHead) {
...
```

And you have removed any need to know about Branch API *and* you are now compatible with any other alternative implementations of things that are built on top.

You should not need to do `branch.getAction` as those actions should be exposed on `job.getAction` automatically as part of the contract of the job returning a `SCMHead` from `SCMHead.HeadByItem.findHead`

Reply:
Thank you, for sharing!
I knew there had to be a better way. Should have explored my options more carefully.

Reply:
In general, if you are needing to depend on branch-api - unless you are doing something specific to branch-api - then you are probably doing things the wrong way.

Reply:
@stephenc I still need the Branch API for `ObjectMetadataAction` and `ContributorMetadataAction`, if I am not mistaken. Or is there another way to get those?

Reply:
Ah, those are from SCM api 👍